src/java.base/share/classes/java/util/regex/Pattern.java

```diff
@@ -2663,7 +2663,11 @@ private CharPredicate clazz(boolean consume) {
                                     right = right.union(clazz(true));
                             } else { // abc&&def
                                 unread();
-                                right = clazz(false);
+                                if(right == null) {
```

Comment:
Missing space after `if`. There will be a unit test as well?

(Please tell me if these comments aren't suitable or relevant, as I'm not involved in the project.)

Reply:
Yep, they are suitable thanks!